### PR TITLE
Various cleanups to save analysis

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**

### CODE REVIEW

File: src/librustc_save_analysis/dump_visitor.rs [hunk 1]
```diff
@@ -92,36 +90,29 @@ pub struct DumpVisitor<'l, 'tcx, 'll> {
...
-    ) -> DumpVisitor<'l, 'tcx, 'll> {
+    ) -> DumpVisitor<'l, 'tcx> {
         let span_utils = SpanUtils::new(&save_ctxt.tcx.sess);
+        let dumper = Dumper::new(save_ctxt.config.clone());
         DumpVisitor {
             tcx: save_ctxt.tcx,
             save_ctxt,
             dumper,
             span: span_utils,
-            cur_scope: CRATE_NODE_ID,
```
Comment (Xanewok) [inline src/librustc_save_analysis/dump_visitor.rs]: Wait, that's weird. So it was unused all along?

Comment (Mark-Simulacrum) [reply src/librustc_save_analysis/dump_visitor.rs]: I think the uses (but not the tracking) was removed in 8a2857e6449c1b90355223367fb4884adb0ba860 by @nrc.

Comment (Xanewok) [reply src/librustc_save_analysis/dump_visitor.rs]: Ah, makes sense. Thanks!

**Review by Xanewok — APPROVED** [review]
That's a welcome cleanup, thanks!

@bors r+ rollup

**Comment by Xanewok** [comment]
Er, does it have to be a regular comment?

@bors r+ rollup

**Pull request merged**